### PR TITLE
Fix #178 and fix #228 - problems in NetworkTest

### LINKED ISSUE #178: Make net.neurons() return Neuron objects instead of strings
Raised from [here](https://github.com/openworm/PyOpenWorm/issues/158#issuecomment-118404848).

@slarson says:
> This could be improved, but at the moment, that is the [behavior in the api](https://pyopenworm.readthedocs.org/en/latest/_modules/PyOpenWorm/network.html#Network.neurons).  Superior behavior would be to return the objects instead of strings as now, but to make sure to implement [`__str__`, `__repr__` and maybe even `unicode`](http://stackoverflow.com/a/1535336/2733572) to ensure that the user can see a meaningful representation of the object when it is inspected interactively or printed out.

### LINKED ISSUE #228: Remove expectedfailure from  test_all_neurons_are_cells
When #178 is done, remove it and close this one.

related: #226

### PR DESCRIPTION
**Pull request by signija**
`net.neurons()` returns Neuron objects now ( #178 ), I removed the expected failure decorator from test_all_neurons_are_cells ( #228 ), and there is a function `net.neuron_names()` with the previous behaviour of net.neurons().

However, I had to add an expected failure decorator to test_neurons in NetworkTest. I tried different things, but I can't get that test to pass with the new implementation of net.neurons(). Running PyOpenWorm and trying out the commands manually works, though. If I do `self.assertEqual()` on `list(net.neurons())` and `[]`, it returns true, so I'm thinking net.neurons() may not work properly within the test, but I can't seem to figure it out.

### CODE REVIEW

**Comment by mwatts15** [comment]
@signija I don't expect the list of neurons to be included for this test. Previously, the test was:
```python
    def test_neurons(self):
        """
        Test that we can access arbitrary Neurons,
        and that they are in the Network
        """
        self.net.neuron(Neuron(name='AVAL'))
        self.net.neuron(Neuron(name='DD5'))
        self.assertTrue('AVAL' in self.net.neurons())
        self.assertTrue('DD5' in self.net.neurons())
```
In this test, the problem is the documentation. It doesn't test arbitrary neurons, but adds neurons to a Network object and then tests that the names of those neurons are returned from that same Network object.

**Comment by signija** [comment]
@mwatts15 Oh, now I see what was so confusing about that test/description initially. That makes sense, thanks!

File: tests/NetworkTest.py [hunk 1]
```diff
@@ -32,15 +32,16 @@ def test_aneuron(self):
         """
         self.assertTrue(isinstance(self.net.aneuron('AVAL'),PyOpenWorm.Neuron))
 
+    @unittest.expectedFailure
```
Comment (slarson) [inline tests/NetworkTest.py]: Great progress!  @signija do you have the info you need to remove the expected failure now before making the final merge?

Comment (signija) [inline tests/NetworkTest.py]: Well, the changes in this test were only caused by trying to solve #178 - I think this PR can be closed and not merged, actually. I don't see anything useful in it other than the observations it allowed us to make.
With the new information, we can try to see what we should do for #178 , change the wording for the original `test_neurons`, and I can fix #228 separately without breaking anything. :smile:

**Pull request closed without merging**